autoload/youcompleteme.vim

```diff
@@ -738,6 +738,20 @@ endfunction
 
 command! YcmShowDetailedDiagnostic call s:ShowDetailedDiagnostic()
 
+function! s:FixIt()
+  " We force recompilation to ensure that the FixIt diagnostic data is not out
+  " of date with respect to the code. Without this we can get unpredicable
+  " results.
+  "
+  " note we do this silently so that we don't echo 2 messages vim. This ensures
+  " that this command doesn't require the user to hit return to dismiss the
+  " Fixit made <N> changes message
+  if s:ForceCompileSilent()
+    py ycm_state.FixIt()
+  endif
+endfunction
+
+command! YcmFixIt call s:FixIt()
```

Comment:
Thinking about it more I think you're right (I didn't recall that the diagnostics were generated from the `OnFileReadyToParse`). And yes, asking other clients' maintainers is a good idea :+1: